lib/internal/modules/esm/resolve.js

```diff
@@ -1058,15 +1058,24 @@ function defaultResolve(specifier, context = {}) {
   }
 
   let parsed, protocol;
-  if (shouldBeTreatedAsRelativeOrAbsolutePath(specifier)) {
-    parsed = URLParse(specifier, parsedParentURL);
-  } else {
-    parsed = URLParse(specifier);
-  }
-
-  if (parsed != null) {
+  try {
+    if (shouldBeTreatedAsRelativeOrAbsolutePath(specifier)) {
+      parsed = new URL(specifier, parsedParentURL);
```

Comment:
Why did we remove URLParse usage? This also reverts the PR that adds it.

Reply:
That's possibly my fault while solving the conflicts. Let me add it back